01-JS-DOM-APIs/Excercise3b/Promise.js

```diff
@@ -0,0 +1,42 @@
+$(document).ready(function () {
+    $(".hidden-section").fadeIn(1000); // do not pay attention to this. It is from a previous excercise. 
+ });
+
+
+function myfunc( method, url, sync ) {
+    
+    var promise = new Promise(function(resolve, reject){
+        var xmlhttp = new XMLHttpRequest(); 
+        xmlhttp.open(method, url, sync);
+        xmlhttp.onreadystatechange = function() { 
+           
+            if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
+                resolve(this.response); 
+            }
+            else {
+                reject(error);
+            }
```

Comment:
this else statement should be an else if, with a condition similar to the one above.
(xmlhttp.readyState === 4 && xmlhttp.status === 200) but instead of comparing with 4 and 200, here you should check for the corresponding error number